File: ios/MidTrim/MidTrim/Domain/UseCases/ResolveExportQualityUseCase.swift
import Foundation
import CoreGraphics

enum ExportQuality: Equatable {
    case free720p
    case paidOriginal(resolution: CGSize)
}

struct ResolveExportQualityUseCase {
    func execute(isPaidUser: Bool, sourceResolution: CGSize) -> ExportQuality {
        if isPaidUser {
            return .paidOriginal(resolution: sourceResolution)
        }
        return .free720p
    }
}

File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/ResolveExportQualityUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.entity.ExportQuality

class ResolveExportQualityUseCase {
    fun execute(isPaidUser: Boolean, sourceWidth: Int, sourceHeight: Int): ExportQuality {
        if (isPaidUser) {
            return ExportQuality.PaidOriginal(sourceWidth, sourceHeight)
        }
        return ExportQuality.Free720p
    }
}